_posts/2016-12-05-haskell-package-checklist.md

```diff
@@ -0,0 +1,162 @@
+  Plus it avoids bit rot by making sure your package will build tomorrow if it builds today.
+
+- **Define your package with hpack.**
+  The default Cabal package file format is custom, tedious, and verbose.
+  hpack uses YAML and avoids unnecessary boilerplate.
+  Stack integrates hpack to automatically convert your `package.yaml` into a `*.cabal` file when necessary.
+
+- **Name your package with kebab-case.**
+  Keep everything lowercase to avoid confusion.
+  You don't want people trying to install `http` when they really meant `HTTP`.
+  Use hyphens to separate words, but keep it short and memorable.
+  Nobody wants to type out `hypertext-transfer-protocol`.
+
+- **Version your package with Semantic Versioning.**
+  Unfortunately Hackage recommends the Package Versioning Policy.
+  The PVP adds ambiguity by using two major version numbers.
+  It also encourages packages to stay on major version 0, which looks bad.
+  Many other languages use Semantic Versioning.
+  SemVer matches how developers generally think about version numbers.
+
+- **Choose a license for your package.**
+  Nobody can use a package without a license.
+  The most popular license for Haskell is BSD 3-Clause, followed by MIT.
+  Whichever license you choose, include the license file in your package (like `LICENSE.markdown`).
+
+- **Write a README.**
+  Most people will familiarize themselves with your package by reading your README.
+  It should describe the problem that your package solves.
+  Be sure to include at least one concrete example in it.
+  And make it look nice by using Markdown (like `README.markdown`).
+
+- **Keep a change log.**
+  Most packages will use Git tags to mark releases.
+  However reading diffs is not an acceptable way for users to discover changes.
+  Put a human-readable summary of changes in the GitHub releases.
+  Link to that from a CHANGELOG file (like `CHANGELOG.markdown`).
+
+- **Write a `synopsis`.**
+  This shows up when searching and viewing your package.
+  Keep it short, imperative, and descriptive.
+  Also write a `description`,
+  which can be pretty much the same as the `synopsis`.
+  For example:
+
+      name:
+        aeson
+      synopsis:
+        encode and decode JSON
+      description:
+        Aeson encodes and decodes JSON.
+
+- **Avoid heavy dependencies.**
+  Only add a dependency if your package needs it to function.
+  Don't include stuff that's just nice to have.
+  For example, you should probably avoid `lens` even though it makes easier to write.
+  In addition to avoiding heavy dependencies,
+  you should avoid having too many dependencies.
+  Think about how long it would take to install your package starting from scratch.
+
+- **Include `extra-source-files`.**
+  If a file is necessary for your package to build, it belongs in `extra-source-files`.
+  This includes files that tests and benchmarks need.
+  You should also include package metadata like your README and change log.
```

Comment:
This should mention that `LICENSE.markdown` does not need to be in `extra-source-files` as long as it's listed as the `license-file`.